### PR TITLE
Added Macho FAT header support

### PR DESCRIPTION
**Pull request by mikem8361**

### CODE REVIEW

File: src/Microsoft.SymbolStore/KeyGenerators/FileKeyGenerator.cs [hunk 1]
```diff
@@ -53,6 +53,7 @@ private IEnumerable<KeyGenerator> GetGenerators()
                 yield return new MinidumpKeyGenerator(Tracer, _file);
                 yield return new ELFFileKeyGenerator(Tracer, _file);
                 yield return new PEFileKeyGenerator(Tracer, _file);
+                yield return new MachOFatHeaderKeyGenerator(Tracer, _file);
                 yield return new MachOFileKeyGenerator(Tracer, _file);
```
Comment (hoyosjs) [inline src/Microsoft.SymbolStore/KeyGenerators/FileKeyGenerator.cs:57]: Are MachO fat and regular file different such that the key generators won't overlap?

Comment (hoyosjs) [reply src/Microsoft.SymbolStore/KeyGenerators/FileKeyGenerator.cs:57]: On a second read, it looks like if it's a flat file, the FAT one would return null and the other one returns the specific stream. Is this correct?

File: src/Microsoft.FileFormats/ELF/ELFProgramHeader.cs [hunk 2]
```diff
@@ -20,6 +15,17 @@ public enum ELFProgramHeaderType : uint
         Phdr = 6
     }
 
+    [Flags]
+    public enum ELFProgramHeaderFlags : uint
```
Comment (hoyosjs) [inline src/Microsoft.FileFormats/ELF/ELFProgramHeader.cs:19]: Is this used anywhere at all?

File: src/Microsoft.FileFormats/ELF/ELFCoreFile.cs [hunk 3]
```diff
@@ -30,6 +28,10 @@ public bool IsValid()
             return _elf.IsValid() && _elf.Header.Type == ELFHeaderType.Core;
         }
 
+        public bool Is64Bit => _elf.Is64Bit;
```
Comment (hoyosjs) [inline src/Microsoft.FileFormats/ELF/ELFCoreFile.cs:31]: Are these properties used?

**Review by hoyosjs — APPROVED** [review]

File: src/Microsoft.SymbolStore/KeyGenerators/FileKeyGenerator.cs (hunk 1 above)
Comment (mikem8361) [reply src/Microsoft.SymbolStore/KeyGenerators/FileKeyGenerator.cs:57]: Yes, the two formats have different header signatures which causes one to fail (return null) and the other to succeed.

File: src/Microsoft.FileFormats/ELF/ELFCoreFile.cs (hunk 3 above)
Comment (mikem8361) [reply src/Microsoft.FileFormats/ELF/ELFCoreFile.cs:31]: They will be.  I needed for this little dump tool that displays info about ELF dumps.

File: src/Microsoft.FileFormats/ELF/ELFProgramHeader.cs (hunk 2 above)
Comment (mikem8361) [reply src/Microsoft.FileFormats/ELF/ELFProgramHeader.cs:19]: This will be used in some upcoming diagnostics extensions work. I wanted to get them in now.

**Pull request merged**